core/src/OpenMP/Kokkos_OpenMP_Parallel.hpp

```diff
@@ -92,53 +102,93 @@ class ParallelFor<FunctorType, Kokkos::RangePolicy<Traits...>, Kokkos::OpenMP> {
       exec_range(const FunctorType& functor, const Member ibeg,
                  const Member iend) {
     const TagType t{};
-#ifdef KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION
-#ifdef KOKKOS_ENABLE_PRAGMA_IVDEP
-#pragma ivdep
-#endif
-#endif
-    for (Member iwork = ibeg; iwork < iend; ++iwork) {
+    KOKKOS_PRAGMA_IVDEP_IF_ENABLED
+    for (auto iwork = ibeg; iwork < iend; ++iwork) {
       functor(t, iwork);
     }
   }
 
+  template <class TagType>
+  inline static
+      typename std::enable_if<std::is_same<TagType, void>::value>::type
```

Comment:
```suggestion
  template <class Enable = WorkTag>
  static std::enable_if_t<std::is_void<WorkTag>::value && std::is_same<Enable, WorkTag>::value>
```
Then at call site `ParallelFor::template exec_work<WorkTag>(m_functor, iWork) -> exec_work(m_functor, iWork);`

Reply:
Yes, not a bad idea to fix that one too